multivac/gather_job_data.py

```diff
@@ -3,12 +3,264 @@
 import glob
 import json
 import os
-import sys
 import re
+import sys
+from pprint import pprint
 
 PROJECT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 sys.path.append(PROJECT_DIR)
 
+version_matcher = re.compile(r"Current runner version: "
+                             r"'(\d*.\d*.\d*)'")
+
+job_was_newer_picked_up = re.compile(r'.*Job is about to start running on the runner.*')
```

Comment:
How about defining possible situations in a declarative way?

```python
situations = [
    {
        "line_re": <...>,
        "counter_name": <...>,
    },
    # <...>
]
```

It'll allow to get rid of the large if-elseif-<...> chain and will ease future extending of the list of situations.

Reply:
@Totktonada I've rebuilt this part, but in fact it's not enough. Some types of failures cannot be detected just from logs alone. Maybe I'll rewrite it with some kind of a sensor class.

Reply:
Thanks! I agree, it is not a full solution, but it structurizes a part of the code better. Most or at least many of failure types are detecteable by a regex on logs.

An idea with a sensor (or sensors) sounds good: it feels close to how I imagine the project architecture.